plugins/modules/keycloak_realm_key.py

```diff
@@ -0,0 +1,427 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2017, Eike Frost <ei@kefro.st>
+# Copyright (c) 2021, Christophe Gilles <christophe.gilles54@gmail.com>
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = '''
+---
+module: keycloak_realm_key
+
+short_description: Allows administration of Keycloak realm keys via Keycloak API
+
+version_added: 7.4.0
+
+description:
+    - This module allows the administration of Keycloak realm keys via the Keycloak REST API. It
+      requires access to the REST API via OpenID Connect; the user connecting and the realm being
+      used must have the requisite access rights. In a default Keycloak installation, admin-cli
+      and an admin user would work, as would a separate realm definition with the scope tailored
+      to your needs and a user having the expected roles.
+
+    - The names of module options are snake_cased versions of the camelCase ones found in the
+      Keycloak API and its documentation at U(https://www.keycloak.org/docs-api/8.0/rest-api/index.html).
+      Aliases are provided so camelCased versions can be used as well.
+
+    - This module is unable to detect changes to the actual cryptographic key after importing it.
+      However, if some other property is changed alongside the cryptographic key, then the key
+      will also get changed as a side-effect, as the JSON payload needs to include the private key.
+      This can be considered either a bug or a feature, as the alternative would be to always
+      update the realm key whether it has changed or not.
```

Comment:
Hmm, this should degrade support for check and diff mode to `partial` since the module is not able to tell whether the private key changed.

I would probably add an "always update private key" option (could be `force`) that users can use if they know the private key is updated, but aren't sure that any other value changes.

Reply:
Marked diff_mode and check_mode as partial.

Reply:
Based on testing - which I need to repeat once more - it *is* possible to update the realm private key. What seems to happen is that on private key update Keycloak notices that the key is now different and creates a new certificate with it. This observation is based on

* Certificate creation timestamp (NotAfter)
* Signing a file with the changed private key and then verifying it with the Keycloak-provided public key

As said in the documentation, the private key (or certificate) could change as a side-effect of changing another parameter as well.

I'll verify the results and then implement the *force* parameter as requested.

Reply:
Actually I made a mistake; this doesn't really affect check mode support, but idempotency.

So instead of adding `support: partial` and `details` for `check_mode`, use `support: full` there, and move the `details:` text to a top-level `notes:` instead. And maybe add `; that is, the module is B(not fully idempotent). You can use O(force=true) to force the module to always update if you know that the private key might have changed.` to the end of the first paragraph.